src/main/java/org/folio/bulkops/util/EntityPathResolver.java

```diff
@@ -35,11 +41,14 @@ public String resolve(EntityType type, BulkOperationsEntity entity) {
         return format("/inventory/view/%s/%s", holding.getInstanceId(), holding.getId());
       }
       case ITEM -> {
-        var item = ((Item) entity);
+        var tenantIdOfEntity = original.getTenant();
+        var item = (Item) entity;
         var holdingId = item.getHoldingsRecordId();
-        var holding =   holdingsClient.getHoldingById(holdingId);
-        var instanceId = holding.getInstanceId();
-        return format("/inventory/view/%s/%s/%s", instanceId, holdingId, item.getId());
+        try (var ignored = new FolioExecutionContextSetter(prepareContextForTenant(tenantIdOfEntity, folioModuleMetadata, folioExecutionContext))) {
+          var holding = holdingsClient.getHoldingById(holdingId);
+          var instanceId = holding.getInstanceId();
```

Comment:
Could you please keep only the holdingsClient invocation inside the try-with-resources block? This will be logically correct and help avoid potential issues in the future. The purpose of this try-with-resources block is to switch the tenant only for the client invocation and then revert it to the original tenant after the block completes.

Reply:
Updated, thanks